### PR TITLE
WIP: Compilation tests

### PR DESCRIPTION
**Pull request by tyrcho**
test to check if Bundler output compiles correctly (#7)

Works fine in IntelliJ.

* [ ] Fails in `sbt test` with a strange exception. 
See https://stackoverflow.com/questions/44217163/dynamic-compilation-with-scala-2-12-from-sbt-test for details

### CODE REVIEW

**Comment by huiwang** [comment]
@tyrcho i stumbed upon a question here about 

http://www.scala-lang.org/api/2.12.0-M4/scala-compiler/scala/tools/reflect/ToolBox.html

https://stackoverflow.com/a/31072625/341307

File: src/test/scala/com/truelaurel/tests/CompilationTest.scala [hunk 1]
```diff
@@ -11,7 +13,9 @@ class CompilationTest extends FlatSpec with Matchers {
   }
 
   it should "reject invalid code" in {
-    compiles("object Toto extends App { println) }") shouldBe false
+    intercept[ToolBoxError]{
+      compiles("object Toto extends App { println) }")
+    }
   }
```
Comment (huiwang) [inline src/test/scala/com/truelaurel/tests/CompilationTest.scala]: @tyrcho here is how to test the exception

File: src/test/scala/com/truelaurel/tests/Compilation.scala [hunk 2]
```diff
@@ -7,14 +7,7 @@ object Compilation {
...
-      toolBox.parse(code)
-      true
-    } catch {
-      case e: Throwable => {
-        e.printStackTrace()
-        false
-      }
-    }
+    toolBox.parse(code)
+    true
```
Comment (huiwang) [inline src/test/scala/com/truelaurel/tests/Compilation.scala]: @tyrcho parse is able to catch errors in the code and it's much much faster than compile

**Comment by tyrcho** [comment]
well done @huiwang  !

**Pull request merged**